viewer/packages/camera-manager/src/Flexible/FlexibleControlsTranslator.ts

```diff
@@ -0,0 +1,66 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+import { OrthographicCamera, PerspectiveCamera, Plane, Raycaster, Vector3 } from 'three';
+import { FlexibleControls } from './FlexibleControls';
+
+/**
+ * @beta
+ * This class handles the translation of the camera.
+ * It tries to translate the camera so the same (x,y, z) point is below the mouse
+ */
+export class FlexibleControlsTranslator {
+  //================================================
+  // INSTANCE FIELDS
+  //================================================
+
+  private readonly _controls: FlexibleControls;
+  private readonly _prevPickedPoint = new Vector3();
+  private readonly _plane = new Plane();
+  private readonly _raycaster = new Raycaster();
+  private _camera: PerspectiveCamera | OrthographicCamera | undefined;
+
+  //================================================
+  // CONTRUCTORS
+  //================================================
+
+  constructor(controls: FlexibleControls) {
+    this._controls = controls;
+  }
+
+  //================================================
+  // INSTANCE METHODS
+  //================================================
+
+  public async initialize(event: PointerEvent): Promise<boolean> {
+    if (!this._controls.getPickedPointByPixelCoordinates) {
+      return false;
+    }
+    const pickedPoint = await this._controls.getPickedPointByPixelCoordinates(event.clientX, event.clientY);
+    if (!pickedPoint) {
+      return false;
+    }
+    this._plane.setFromNormalAndCoplanarPoint(this._controls.cameraVector.getEndVector(), pickedPoint);
+    this._camera = this._controls.camera.clone();
+    this._prevPickedPoint?.copy(pickedPoint);
+    return true;
+  }
+
+  public translate(event: PointerEvent): boolean {
+    if (!this._camera) {
+      return false;
+    }
+    const normalizedCoords = this._controls.getNormalizedPixelCoordinates(event);
+    this._raycaster.setFromCamera(normalizedCoords, this._camera);
```

Comment:
I could not get where this function is used, hope it is not called on `MouseMove` or continuous frames as `raycast` would cause us performance issues (I guess you already know if 😃  )

Reply:
It is used in FlexibleControls.startMouse. Why should this use long time. I'm nor ray casting anything. Just get the ray in order to get the direction from the camera.